### PR TITLE
fix all warnings, set high warning level

### PR DESCRIPTION
**Pull request by NPIPHI**
Fix all the warnings so silsim will compile on -Wall -Wextra -Wpedantic with no warnings

### CODE REVIEW

File: src/Sensor.cpp [hunk 1]
```diff
@@ -114,8 +114,7 @@ void Barometer::update_data(double tStep) {
         new_data_ = true;
 
         if (inject_noise_) {
-            noise_ = normal_dist_(generator_);
-            data_ += noise_;
+            data_ = normal_dist_(generator_);
```
Comment (AyberkY) [inline src/Sensor.cpp:117]: Hmm. Why did you get ride of the `noise_` variable for the Barometer? I see it was deleted from `Sensor.h`.

File: includes/Sensor.h [hunk 2]
```diff
@@ -106,16 +106,14 @@ class Barometer : public Sensor {
...
-    double get_data();
+    void update_data(double tStep) override;
+    double get_data() override;
 
     void set_constant_bias(double bias) { bias_ = bias; };
 
    private:
     double data_;  // The sensor's current reading
 
-    double noise_;  // Noise value to be added to measurement
```
Comment (AyberkY) [inline includes/Sensor.h:117]: Why did you delete this?

**Review by AyberkY — CHANGES_REQUESTED** [review]
You might have deleted `noise_` for the Barometer class by accident 🤔

File: includes/Sensor.h (hunk 2 above)
Comment (NPIPHI) [reply includes/Sensor.h:117]: I don't think there needs to me a variable to hold the noise when it is only ever used to add to the measured data. I've put it back for now though.

**Review by AyberkY — APPROVED** [review]
lgtm
